src/bridge/cockpitpipechannel.c

```diff
@@ -483,3 +510,46 @@ cockpit_pipe_channel_open (CockpitTransport *transport,
   json_object_unref (options);
   return channel;
 }
+
+static void
+internal_fd_free (gpointer data)
+{
+  gint fd = GPOINTER_TO_INT (data);
+
+  close (fd);
+}
+
+const gchar *
+cockpit_pipe_channel_add_internal_fd (gint fd)
+{
+  /* We are not multi-threaded. Also don't make this look like normal fd numbers */
+  static guint64 unique = 911111;
```

Comment:
No, the fd might be closed and reused before the client calls `cockpit.channel()`, which would give it the wrong fd.